webapp/src/app/modules/publication/models/paper.model.ts

```diff
@@ -19,43 +18,16 @@ class AssetFile implements IAsset {
     if(file) {

```

Comment:
Here you should inject the services File and Hash (or Crypto) in order to be using by this class. You can do it on this way: https://stackoverflow.com/questions/41432388/how-to-inject-service-into-class-not-component